### PR TITLE
feat(sheet-overlay): close overlay on esc key press

### PR DESCRIPTION
**Pull request by DmitiryPotychkin**
- add option to close sheet overlay with esc key

### CODE REVIEW

File: projects/components/src/overlay/sheet/sheet-overlay.component.ts [hunk 1]
```diff
@@ -46,6 +46,10 @@ export class SheetOverlayComponent {
   public readonly rendererInjector: Injector;
   public visible: boolean = true;
 
+  @HostListener('document:keydown.escape', ['$event']) onKeydownHandler() {
+    this.close();
```
Comment (aaron-steinfeld) [inline projects/components/src/overlay/sheet/sheet-overlay.component.ts:50]: please move below constructor and add visibility/type declaration. Also, can we add this to modal too?

File: projects/components/src/modal/modal-container.component.ts [hunk 2]
```diff
@@ -72,6 +74,11 @@ export class ModalContainerComponent {
     this.rendererContext = this.rendererInjector.get(MODAL_DATA, {});
   }
 
+  @HostListener('document:keydown.escape', ['$event'])
+  public onKeydownHandler(): void {
+    if (this.closeOnEscape) this.close();
```
Comment (aaron-steinfeld) [inline projects/components/src/modal/modal-container.component.ts:79]: braces

File: projects/components/src/modal/modal-container.component.ts [hunk 3]
```diff
@@ -59,6 +60,7 @@ export class ModalContainerComponent {
     this.modalTitle = config.title ?? '';
     this.size = config.size;
     this.isComponentModal = !(config.content instanceof TemplateRef);
+    this.closeOnEscape = config.closeOnEscapeKey === undefined ? true : config.closeOnEscapeKey;
```
Comment (aaron-steinfeld) [inline projects/components/src/modal/modal-container.component.ts:63]:
```suggestion
    this.closeOnEscape = config.closeOnEscapeKey ?? true;
```

File: projects/components/src/popover/popover.ts [hunk 4]
```diff
@@ -6,6 +6,7 @@ export interface PopoverOptions<TData = never> {
   position: PopoverPosition;
   parentInjector?: Injector;
   backdrop?: PopoverBackdrop;
+  closeOnEscapeKey?: boolean;
```
Comment (aaron-steinfeld) [inline projects/components/src/popover/popover.ts:9]: not needed in popover

**Review by aaron-steinfeld — APPROVED** [review]

File: projects/components/src/overlay/sheet/sheet.ts [hunk 5]
```diff
@@ -7,6 +7,7 @@ export interface SheetOverlayConfig<TData = unknown> extends OverlayConfig {
   size: SheetSize;
   data?: TData;
   position?: PopoverFixedPositionLocation.Right | PopoverFixedPositionLocation.RightUnderHeader;
+  closeOnEscapeKey?: boolean;
```
Comment (anandtiwary) [inline projects/components/src/overlay/sheet/sheet.ts:10]: nit: we should add a comment about the default value for these properties.

**Review by anandtiwary — APPROVED** [review]

**Pull request merged**